tests/test_target_space.py

```
@@ -273,6 +293,54 @@ def test_res():
+    ]
+    assert len(space.res()) == 4
+    assert space.res() == expected_res
+
+
+def test_res_categorical_with_constraints() -> None:
+    PBOUNDS = {"p1": (0, 10), "p2": ["a", "b", "c"]}
+
+    def _f(p1: float, p2: str) -> float:
+        return p1 + len(p2)
+
+    space = TargetSpace(_f, PBOUNDS, constraint=NonlinearConstraint(lambda p1, p2: p1 - 2, 0, 5))
```

Ruff (0.15.4)

[warning] Unused lambda argument: `p2`

(ARG005)